resources/aws/kms.go

```diff
@@ -19,6 +19,10 @@ func (kk KMSKey) fullAlias() string {
 }
 
 func (kk *KMSKey) CreateOrFail() error {
+	if kk.Name == "" {
+		return microerror.MaskAny(kmsKeyAliasEmptyError)
+	}
```

Comment:
The name [comes from `cluster.Name`](https://github.com/giantswarm/aws-operator/blob/a58abe3eb816bf1fd0ce8804ce95a5f635ac082b/service/create/service.go#L219).

It is unlikely that we didn't encounter an error yet if that's the case, but I think it makes sense to enforce it here, since this is what the KMS key cares about.